libpod/container_api.go

```diff
@@ -102,6 +102,8 @@ func (c *Container) Init() (err error) {
 		return errors.Wrapf(err, "unable to generate hostname file for container")
 	}
 
+	// Set the hostname in the env variables
+	c.config.Spec.Process.Env = append(c.config.Spec.Process.Env, fmt.Sprintf("HOSTNAME=%s", c.Spec().Hostname))
```

Comment:
Use c.config.Spec instead of c.Spec() here to avoid a copy of the struct

Reply:
ok fixed